osu.Game/OsuGame.cs

```diff
@@ -609,6 +609,9 @@ protected override void LoadComplete()
 
             loadComponentSingleFile(screenshotManager, Add);
 
+            // dependency on notification overlay
+            loadComponentSingleFile(CreateUpdateManager(), Add, true);
```

Comment:
Is there any particular reason for this location of this load?

Reply:
UpdateManager depends on NotificationOverlay. SettingsOverlay has UpdateSettings which depends on UpdateManager, so it has to be "sandwiched".

Reply:
The `UpdateManager -> SettingsOverlay` dependency should be annotated with an inline comment for posterity in that case, albeit I don't really like how interdependent everything is starting to become.